src/OrchardCore.Modules/OrchardCore.Media/Services/ViewMediaFolderAuthorizationHandler.cs

```diff
@@ -0,0 +1,228 @@
+                var contentItemId = attachedMediaPathParts.Length > 1 ? attachedMediaPathParts[1] : null;
+                var contentItem = !string.IsNullOrEmpty(contentItemId) ? await _contentManager.GetAsync(contentItemId) : null;
+
+                // Disallow if content item is not found or allowed
+                if (contentItem is not null)
+                {
+                    await AuthorizeAsync(context, requirement, Contents.CommonPermissions.ViewContent, contentItem);
+                }
+            }
+        }
+
+        private async Task AuthorizeUsersFolderAsync(AuthorizationHandlerContext context, PermissionRequirement requirement, string path)
+        {
+            // We need to allow the _Users folder for own media access too. If someone uploads into this folder, we are screwed.
+            Permission permission;
+            if (path.IndexOf(PathSeparator) < 0)
+            {
+                permission = SecureMediaPermissions.ViewOwnMedia;
+            }
+            else
+            {
+                permission = SecureMediaPermissions.ViewOthersMedia;
+
+                var userFolderName = _userAssetFolderNameProvider.GetUserAssetFolderName(context.User);
+                if (!string.IsNullOrEmpty(userFolderName))
+                {
+                    var userOwnFolder = EnsureTrailingSlash(_fileStore.Combine(_usersFolder, userFolderName));
+
+                    if (IsAuthorizedFolder(userOwnFolder, path) || IsDescendantOfAuthorizedFolder(userOwnFolder, path))
+                    {
+                        permission = SecureMediaPermissions.ViewOwnMedia;
+                    }
+                }
+            }
+
+            await AuthorizeAsync(context, requirement, permission);
+        }
+
+        private async Task AuthorizeAsync(AuthorizationHandlerContext context, PermissionRequirement requirement, Permission permission, object resource = null)
+        {
+            var authorizationService = _serviceProvider.GetService<IAuthorizationService>();
+            if (await authorizationService.AuthorizeAsync(context.User, permission, resource))
+            {
+                MarkAsSecure(context.User);
+                context.Succeed(requirement);
+            }
+            else
+            {
+                // Note: We don't want other authorization handlers to succeed the requirement. This would allow access to the
+                // users and attached media field folders, e.g. if the anonymous role has the "ViewMedia" permission set.
+                context.Fail(new AuthorizationFailureReason(this, "View media permission not granted"));
+            }
+        }
+
+        private void MarkAsSecure(ClaimsPrincipal user)
+        {
+            // If anonymous access is allowed, also allow default caching in the browser.
+            if(user.Identity?.IsAuthenticated ?? false)
+                _httpContextAccessor.HttpContext.Items["IsSecureMedia"] = "true";
+        }
```

Comment:
This marks all Media files secure when the user is authenticated. I don't know what the best solution here would be, but doing this in a blanket way is something we definitely don't want: even if your site utilizes Secure Media for some files, you can very well have the rest (most) of your Media non-secured, and that should be possible to cache for authenticated users too.

Reply:
I changed this now. If a particular path is accessible for anonymous users, default browser caching is now used.

Reply:
Great!